Add test cases for File Block (#57)
wordpress-mobile/test-cases: test-cases/gutenberg/file.md

# File Block - Test Cases
##### TC001

**Upload completes successfully from the device library**

**Steps**
- Tap on Choose a file on the Media Placeholder. 
- Tap option "Choose from device" and choose a file. 

**Expected Behavior**
- Expect to see the upload indicator and the Download button should be dimmed.
- Expect to not see a link or replace buttons in the toolbar.

![Upload pending](../resources/local-file-upload-pending.png)
- Expect to see file's name and an active Download button once the upload is complete.  
- Expect the link and replace buttons to appear in the toolbar

![Upload successful](../resources/local-file-upload-successful.png)

--------------------------------------------------------------------------------

##### TC002

> **Upload completes successfully from the WordPress Media library**

Reviewer: Nitpick: Adding from WordPress Media Library is technically not an Upload. It's just inserted.

Author: Fixed in bfad517 along with placeholder changes we had discussed.

**Steps**
- Tap on Choose a file on the Media Placeholder. 
- Tap option "WordPress Media Library" and choose a file. 

**Expected Behavior**

> - Expect to see file's name  (without link)  and an active Download button.  

Reviewer: What does it mean `without link`?

Author: Removed here bd426a5

-   You should be asked to retry.

**Expected Behavior**

![Upload retry](../resources/local-file-upload-retry.png)

**Next Step**
-   Tap retry

**Expected Behavior**
-   The upload should continue.

**Alternative Step**
-   Tap Remove upload.

**Expected Behavior**
-   The placeholder should now be visible and the Download button and link should be cleared.
--------------------------------------------------------------------------------
##### TC004

**Stop the file upload before it completes**

**Steps**

-   Tap on Choose a file on the Media Placeholder or edit button in the toolbar.
-   Tap option ‘Choose from device’ and choose a file.
-   While the upload is in progress tap on the File Block.

**Expected Behavior**

-   You should see the below message:
![Upload stopped](../resources/local-file-upload-stop.png)

**Next Step**

-   Tap Yes

**Expected Behavior**

-   You should see that File block is cleared and the placeholder component is now visible.
![Upload placeholder](../resources/local-file-upload-placeholder.png)

--------------------------------------------------------------------------------
##### TC005

**Close the post with an ongoing file upload**

> 

Reviewer: I would also add the test case of closing the editor and opening it again while a file is being updated.
Expected:
- The block should detect the ongoing upload and show the progress bar.

Author: Done 30d7541

**Expected Behavior**

-   Verify that you see a new filename.

--------------------------------------------------------------------------------
##### TC007

**Change the alignment in the File Block**

**Step**
- Press on the Alignment toolbar button.

<kbd><img src="../resources/file-alignment.png" width="320"></kbd>

- Choose one of the options. For example, Align Right.

**Expected Behavior**

- Check to ensure the UI represents the alignment accordingly.
Note: Align left, Wide width and no option selected won't have visual differences.

![Alignment](../resources/file-alignment-right.png)

--------------------------------------------------------------------------------
##### TC008

The test cases below apply to both the Block Settings icon and the Link toolbar icon. The only differentiator is the Block Settings contain a "Show download button" option while the Link settings via the toolbar does not. 

**The Link To Block Setting should open**

**Steps**

- Press the Block settings or Link Settings icon.
- Press the Link to icon. 

**Expected Behavior**

- Two different options should appear: Media file and Attachment Page
- Media File: The link should be set to the file itself.

> - Attachment page: The link should be set to the item page.

Reviewer: I would add a step to check the preview site that the proper link was set and it is not broken.
Even though there's that known issue about recently uploaded files that will change their attachment url after autosaving on mobile.

Author: Sounds good!

**Steps**

- Press the Block settings or Link Settings icon.
- Switch ON the option - Open link in new tab

**Expected Behavior**

- On HTML mode there should be the attributes `target="_blank" rel="noreferrer noopener"` in the `<a>` tag.

--------------------------------------------------------------------------------
##### TC0010

**Show/hide download button (Block Settings only)**

**Steps**

- Press the Block settings or Link Settings icon.
- Switch off the "Show download button" option.

**Expected Behavior**

- The Download button should disappear

--------------------------------------------------------------------------------
##### TC0011

**Copy file URL**

**Steps**

- Press the Block settings or Link Settings icon.
- Press on the Copy file URL option.
- The button title should change to Copied!

**Expected Behavior**

- The content should be in the device's clipboard.
- Paste the link from the pasteboard in any other field and it should appear.

--------------------------------------------------------------------------------
##### TC012

**Close/Re-open post with an ongoing file upload**

**Steps**

-   Add a File block and insert a file from your device.

> -   While there’s an ongoing upload, save the post as a draft.

Reviewer: I think we should note that closing the editor is a necessary step here too.

```suggestion
-   While there’s an ongoing upload, close the editor by saving the post as draft.
```
